### ISSUE TITLE
Handling YZ divergence w/ batching/queues: Flush queues before doing KV/YZ AAE sync [JIRA: RIAK-2252]

### ISSUE THREAD

**Issue by zeeshanlakhani** [basho/yokozuna#580]
With the new queuing code we are seeing elevated levels of YZ repair as there is a longer lag between KV data arriving and being indexed.  At sync time, snapshot (update_tree) the KV tree, flush the YZ queue, snapshot (update_tree) YZ tree, exchange.

**Comment by jvoegele** [comment]
Write a riak_test to verify startup/shutdown behavior.

**Comment by zeeshanlakhani** [comment]
as per discussion/walkthrough w/ @fadushin (related need: https://github.com/basho/yokozuna/issues/582).

``` erlang
 update_request(riak_kv_index_hashtree, KVTree, Index, IndexN), %% runs update_tree kv
*YZMOD*:flush_queues %% Flush queues call related to https://github.com/basho/yokozuna/issues/582
 update_request(yz_index_hashtree, YZTree, Index, IndexN), %% runs update_tree yz
...
```

**Comment by JeetKunDoug** [comment]
@zeeshanlakhani assuming we can close this now.

**Comment by Basho-JIRA** [comment]
Fixed in 2.0.7 

**_[posted via JIRA by Fred Dushin]**_
